dialer_test.go

```diff
@@ -644,14 +694,41 @@ func TestConnectedState(main *testing.T) {
 [DEBUG] connection closed
 `, l.Logs())
 	})
+
+	main.Run("UrlsIterEqualUrlsOrder", func(t *testing.T) {
+		defer goleak.VerifyNone(t)
+		ctrl := gomock.NewController(t)
+		defer ctrl.Finish()
+
+		urlsPool := []string{
+			"the.first.url",
+			"the.second.url",
+			"the.last.url",
+		}
+		wantedIndex := 0
+		dialSub := func(url string, config amqp.Config) (amqpextra.AMQPConnection, error) {
+			assert.Equal(t, url, urlsPool[wantedIndex])
+			wantedIndex++
+
+			return nil, errors.New("the error")
+		}
+
+		dialer, err := amqpextra.NewDialer(
+			amqpextra.WithURL(urlsPool[0], urlsPool...),
+			amqpextra.WithAMQPDial(dialSub),
+		)
+
+		require.NoError(t, err)
+
+		dialer.Close()
```

Comment:
you have to wait here for sometime, cuz now it exists after first dial. 

So, set some retryPeriod and wait for 2.5 of its time.

Reply:
+